src/Forms/ConfirmedPasswordField.php

```diff
@@ -155,6 +181,43 @@ public function __construct(
         $this->setValue($value);
     }
 
+    /**
+     * Endpoint to provide feedback for the current and required level of password strength
+     */
+    public function strength(HTTPRequest $request): HTTPResponse
+    {
+        $response = HTTPResponse::create();
+        $json = json_decode($request->getBody() ?? '', true);
+        if (!$json || !array_key_exists('password', $json) || !$request->isPOST()) {
+            $response->setStatusCode(400);
+            return $response;
+        }
+        $password = $json['password'];
+        $validator = PasswordValidator::create();
+        $strength = $validator->evaluateStrength($password);
+        $strengthLabel = $validator->getStrengthLabel($strength);
```

Comment:
```suggestion
        $strength = PasswordStrengthValidator::estimateStrength($password);
        $strengthLabel = $validator->getStrengthLabel($strength);
```

[Symfony gives us this funcitonality](https://symfony.com/doc/current/reference/constraints/PasswordStrength.html#customizing-the-password-strength-estimation).

We should also move the `getStrengthLabel()` either into this class or specifically only in `EntropyPasswordValidator` (but probably here since this is where we're using it). We don't have to implement any of this in our `PasswordValidator` or `EntropyPasswordValidator` classes.

Reply:
Updated